### PR TITLE
allow negative pr number to bypass git fetch errors

### PR DESCRIPTION
**Pull request by JudeNiroshan**
### Allow pipeline executor to enter negative pr number

workaround which enable to execute the `taxonomy_base` to `empty`

 It will validate the entire taxonomy by performing a diff against an empty base.
Ref: https://github.com/instructlab/instructlab?tab=readme-ov-file#-list-and-validate-your-new-data

### CODE REVIEW

**Review by tumido — APPROVED** [review]
LGTM 🚀

File: sdg/components.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ def git_clone_op(
             f"git clone {repo_url} {taxonomy.path} && cd {taxonomy.path} && "
             + f'if [ ! -z "{repo_branch}" ]; then '
             + f"git fetch origin {repo_branch} && git checkout {repo_branch}; "
-            + f'elif [ ! -z "{repo_pr}" ]; then '
+            + f'elif [ ! -z "{repo_pr}" and {repo_pr} -gt 0 ]; then '
```
Comment (leseb) [inline sdg/components.py:22]:
```suggestion
            + f'elif [ ! -z "{repo_pr}" && {repo_pr} -gt 0 ]; then '
```
`and`? in bash? :)

**Review by leseb — CHANGES_REQUESTED** [review]

File: sdg/components.py (hunk 1 above)
Comment (leseb) [reply sdg/components.py:22]: Also, a good opportunity to switch from `! -z` with `-n` :)

File: sdg/components.py [hunk 2]
```diff
@@ -43,8 +43,8 @@ def sdg_op(
     endpoint = getenv("endpoint")
     client = openai.OpenAI(base_url=endpoint, api_key=api_key)
 
-    taxonomy_base = "main" if repo_branch or repo_pr else "empty"
-
+    taxonomy_base = "main" if repo_branch or repo_pr > 0 else "empty"
+
```
Comment (leseb) [inline sdg/components.py:47]: trailing space

File: pipeline.yaml [hunk 3]
```diff
@@ -467,7 +467,7 @@ deploymentSpec:
         - -c
         - "\nif ! [ -x \"$(command -v pip)\" ]; then\n    python3 -m ensurepip ||\
           \ python3 -m ensurepip --user || apt-get install python3-pip\nfi\n\nPIP_DISABLE_PIP_VERSION_CHECK=1\
-          \ python3 -m pip install --quiet --no-warn-script-location 'kfp==2.9.0'\
+          \ python3 -m pip install --quiet --no-warn-script-location 'kfp==2.8.0'\
```
Comment (leseb) [inline pipeline.yaml:470]: Does this have anything to do with allowing negative number? If not, could you keep in a separate OR or at least a separate commit? Thanks!

File: sdg/components.py (hunk 1 above)
Comment (JudeNiroshan) [reply sdg/components.py]: c16d456ae504da7a5c53fb8573a784fb1e5be6c8

File: sdg/components.py (hunk 2 above)
Comment (JudeNiroshan) [reply sdg/components.py]: 0f3ce38d013887095bb109a7a5f3eec08c007b6f

File: pipeline.yaml (hunk 3 above)
Comment (JudeNiroshan) [reply pipeline.yaml]: 3fdfb3aacbbcb0da231c6029f943f7488ab1c476

**Comment by leseb** [comment]
This PR https://github.com/redhat-et/ilab-on-ocp/pull/38 switched from 2.9 to 2.8 why are we switching again?

**Comment by JudeNiroshan** [comment]
> This PR #38 switched from 2.9 to 2.8 why are we switching again?

It looks like in that PR they forgot to commit pipenv.lock. As you can see in the main branch, we are still using [kfp 2.9.0](https://github.com/redhat-et/ilab-on-ocp/blob/8a0eec425a9dc77b198eada4d9d0836896fead93/Pipfile.lock#L245), but in the latest generated yaml, it is on 2.8.0.

**Review by cooktheryan — APPROVED** [review]

File: sdg/components.py [hunk 4]
```diff
@@ -43,7 +43,7 @@ def sdg_op(
     endpoint = getenv("endpoint")
     client = openai.OpenAI(base_url=endpoint, api_key=api_key)
 
-    taxonomy_base = "main" if repo_branch or repo_pr else "empty"
+    taxonomy_base = "main" if repo_branch or repo_pr > 0 else "empty"
```
Comment (leseb) [inline sdg/components.py:46]: This breaks since repo_branch is string.
